src/main/java/com/igot/cb/discussion/service/impl/DiscussionServiceImpl.java

```diff
@@ -418,32 +394,39 @@ public ApiResponse updateUpVote(Map<String, Object> upVoteData, String token) {
                     response.setResponseCode(HttpStatus.INTERNAL_SERVER_ERROR);
                     return response;
                 }
-                long newVoteCount = existingVoteCount + vote;
-                discussionData.put(Constants.VOTE_COUNT, newVoteCount);
+                if (voteType.equals(Constants.UP)) {
+                    discussionData.put(Constants.UP_VOTE_COUNT, existingUpVoteCount + 1);
+                } else {
+                    discussionData.put(Constants.DOWN_VOTE_COUNT, existingDownVoteCount + 1);
+                }
             } else {
                 Map<String, Object> userVoteData = existingResponseList.get(0);
                 if (userVoteData.get(Constants.VOTE_TYPE).equals(voteType)) {
                     createErrorResponse(response, String.format(Constants.USER_ALREADY_VOTED, voteType), HttpStatus.ALREADY_REPORTED, Constants.FAILED);
                     return response;
                 }
+
                 Map<String, Object> updateAttribute = new HashMap<>();
                 updateAttribute.put(Constants.VOTE_TYPE, voteType);
                 Map<String, Object> compositeKeys = new HashMap<>();
                 compositeKeys.put(Constants.USER_ID_RQST, userId);
-                compositeKeys.put(Constants.DISCUSSION_ID_KEY, upVoteData.get(Constants.DISCUSSION_ID));
+                compositeKeys.put(Constants.DISCUSSION_ID_KEY, discussionId);
+
                 Map<String, Object> result = cassandraOperation.updateRecordByCompositeKey(Constants.KEYSPACE_SUNBIRD, Constants.USER_DISCUSSION_VOTES, updateAttribute, compositeKeys);
                 if (!result.get(Constants.RESPONSE).equals(Constants.SUCCESS)) {
                     createErrorResponse(response, Constants.FAILED_TO_VOTE, HttpStatus.INTERNAL_SERVER_ERROR, Constants.FAILED);
                     return response;
                 }
-                if (userVoteData.get(Constants.VOTE_TYPE).equals(Constants.UP) && voteType.equals(Constants.DOWN)) {
-                    vote = -2L;
+
+                if (voteType.equals(Constants.UP)) {
+                    discussionData.put(Constants.UP_VOTE_COUNT, existingUpVoteCount + 1);
+                    discussionData.put(Constants.DOWN_VOTE_COUNT, existingDownVoteCount - 1);
                 } else {
-                    vote = 2L;
+                    discussionData.put(Constants.UP_VOTE_COUNT, existingUpVoteCount - 1);
+                    discussionData.put(Constants.DOWN_VOTE_COUNT, existingDownVoteCount + 1);
```

Comment:
this is not correct... 
since we have two variables -- each upVotes and downVotes needs to be measured properly...